packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -39,6 +39,23 @@ import 'xcode_build_settings.dart';
 import 'xcode_debug.dart';
 import 'xcodeproj.dart';
 
+const String kJITCrashFailureMessage =
+    'Crash occurred when compiling unknown function in unoptimized JIT mode in unknown pass';
+
+const String kJITCrashFailureInstructions = '''
+════════════════════════════════════════════════════════════════════════════════
+A change to iOS has caused a temporary break in Flutter's debug mode on
+physical devices.
+See https://github.com/flutter/flutter/issues/163984 for details.
+
+In the meantime, we recommend these temporary workarounds:
+
+* When developing with a physical device, use one running iOS 18.3 or lower.
```

Comment:
This is better than nothing but likely too late since they've already upgraded. During betas, they'll have the ability to (painfully) return to 18.3, but once final is out and they stop signing 18.3 this will be too late.

Consider adding a TODO here to update this message as we learn more, and perhaps get more aggressive and publish a (smaller) warning even for those on earlier releases too.

Reply:
I think I'll file a Github issue to follow up rather than a TODO

Reply:
sgtm! Could do both -- TODOs require a bug, and makes it easy to grep for the spot to fix later on.